Mobile/ShopLens/Droid/Activities/ShoppingListActivity.cs

```diff
@@ -106,24 +111,36 @@ public void OnResults(Bundle results)
             var matches = results.GetStringArrayList(SpeechRecognizer.ResultsRecognition);
             if (matches.Count > 0)
             {
-                Regex addProductRegex = new Regex(@"^" + ConfigurationManager.AppSettings["CmdAddCartList"] + @"(?:\s\w+)+");
+                string cmdAddProduct = ConfigurationManager.AppSettings["CmdAddCartList"];
+                Regex addProductRegex = new Regex(@"^" + cmdAddProduct + @"(?:\s\w+)+");
 
                 if (matches[0] == voiceListCmd)
                 {
+                    int voicerAwaitTime = int.Parse(ConfigurationManager.AppSettings["VoicerPauseTime"]);
                     string endMessage = "Voicing of shopping list complete.";
 
-                    foreach (string item in items)
+                    Task.Run(() =>
                     {
-                        SpeakOut(item);
-                        Thread.Sleep(int.Parse(ConfigurationManager.AppSettings["VoicerPauseTime"]));
-                    }
+                        foreach (string item in items)
+                        {
+                            SpeakOut(item);
+                            Thread.Sleep(voicerAwaitTime);
+                        }
 
-                    SpeakOut(endMessage);
+                        SpeakOut(endMessage);
+
+                    }).ContinueWith((t) =>
+                    {
+                        if (t.IsFaulted)
+                        {
+                            System.Diagnostics.Debug.WriteLine(t.Exception);
+                        }
+                    }); 
                 }
 
                 if (addProductRegex.IsMatch(matches[0]))
                 {
-                    string itemToAdd = ProductNameHelper.GetProductNameFromString(matches[0]).FirstCharToUpper();
+                    string itemToAdd = string.Join("", matches[0].Skip(cmdAddProduct.Length + 1)).FirstCharToUpper();
```

Comment:
"👆 Va čia pagalvokite labai gerai, kodėl čia negerai, pagalvokite... " 🏃

Hint: why do you need the `Join`? If you make matches[0] a variable maybe it will be more clear

Reply:
That running man emote looked so different on my mobile phone.

Reply:
Where did you get that quote from? Seems too cliche to be original.